src/jenkins_job_insight/main.py

```
@@ -508,33 +525,95 @@ async def lifespan(app: FastAPI):
+        # 3. Fall back to jji_username cookie (regular users)
+        if not username:
+            username = request.cookies.get("jji_username", "")
+
         request.state.username = username
+        request.state.is_admin = is_admin
+        request.state.role = "admin" if is_admin else "user"
+
+        # Track regular user activity (non-admin, has username)
+        if username and not is_admin:
+            # Fire and forget — don't block the request
+            asyncio.create_task(_safe_track_user(username))
```

Ruff (0.15.10)

[warning] Store a reference to the return value of `asyncio.create_task`

(RUF006)